geochem_inverse_optimize.py

```diff
@@ -198,13 +199,38 @@ def solve(
 
         return downstream_preds, upstream_preds
 
+    def solve_montecarlo(
+        self,
+        observation_data: ElementData,
+        relative_error: float,
+        num_repeats: int,
+        regularization_strength: float,
+        solver: str = "gurobi",
+    ):
+        predictions_down_mc = defaultdict(list)
+        predictions_up_mc = defaultdict(list)
+        for i in range(num_repeats):
+            observation_data_resampled = {
+                sample: value * np.random.normal(loc=1, scale=relative_error / 100)
+                for sample, value in observation_data.items()
+            }
+            element_pred_down, element_pred_upstream = self.solve(
+                observation_data=observation_data_resampled,
+                solver=solver,
+                regularization_strength=regularization_strength,
+            )  # Solve problem
+            for sample_name in element_pred_down.keys():
+                predictions_down_mc[sample_name] += [element_pred_down[sample_name]]
+                predictions_up_mc[sample_name] += [element_pred_upstream[sample_name]]
+            i += 1
```

Comment:
This seems incorrect.

Reply:
The `i+=1' isn't need so I removed it. 
I checked that this still produces the expected outcomes and it does.